### DISCUSSION TITLE
How to use an input datepicker in custom module extension

### DISCUSSION THREAD

**Discussion by tannontawat** [directus/directus#13631] [Archive]
I want to use an input date picker in my custom module extension.
How to implement it ??

Many thanks for considering my request.

**Comment by rijkvanzanten** [comment]
You can use the core interfaces by using `<interface-datetime :value="val" @input="val = $event" />` All options for the interface are passed in through props, like `<interface-datetime include-seconds 
/>`

**Reply by tannontawat** [reply]
I add the interface like this
`<interface-datetime v-model="versionTo" :includeSeconds="true" :type="datePickerType" :value="versionTo" @input="versionTo = $event" />`  type is `dateTime`. so, it not change at my model and the datepicker is still show on the screen.

What should I do?
![image](https://user-images.githubusercontent.com/32186802/171217135-f975e86a-a5e7-475e-9158-1ecf79957881.png)

Thanks.

**Reply by rijkvanzanten** [reply]
I don't know what you mean by

> so, it not change at my model and the datepicker is still show on the screen.

**Reply by tannontawat** [reply]
I have a model name `versionTo` and I pass props `:includeSeconds="true", :type="dateTime"` when I click the datepicker and choose the date, it not change the value at my model and the datepicker is not hidden from the screen. I don't sure what I do wrong?
![image](https://user-images.githubusercontent.com/32186802/171222558-cc38a92c-a30b-41d7-a7ac-58a6f6dd7e95.png)
![image](https://user-images.githubusercontent.com/32186802/171222677-5f0d4f7e-ea35-4683-83d8-76547d55c19d.png)

**Reply by rijkvanzanten** [reply]
Hard to tell without the full source code of the component.. There could be a range of reasons why it's not quite working as expected 🤔

**Reply by tannontawat** [reply]
Yesterday, I tried to do that. My solution is to create the method to set the model's value like this `<interface-datetime :includeSeconds="true" :type="datePickerType" :use24="true" :value="versionFrom" @input="setVersionFrom($event)" />`.
So it works. ^^

Thanks for help.
